src/code/sys_cmpdma.c

```diff
@@ -1,32 +1,117 @@
 #include "global.h"
 #include "system_malloc.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cmpdma/func_80178750.s")
+typedef struct {
+    union {
+        u32 dmaWord[2];
+        u32 dataStart;
+        u32 dataSize;
+        struct {
+            u32 start;
+            u32 end;
+        } offset;
+    };
+} CmpDmaBuffer;
```

Comment:
To be honest I'm not sure how I feel about the union. While it does make things clear in the code below, it just adds a lot of complexity that I'm not sure is worth it. I think I still prefer an array of length 2 with comments explaining what the buffer is being used for. However, I am willing to keep the union.

Reply:
eh, I think it's fine as the complexity is well contained within the file.

Reply:
Personally I'm fine with the union.